### PR TITLE
Scheme/Racket corrections

### PR DESCRIPTION
**Pull request by dented42**
Renamed 'Scheme' to 'Racket'.
Renamed 'Guile' to 'Scheme'.
Made racket source files (including the example) use `#lang racket/base` instead of `#lang racket`, which shaves off ~500ms.

### CODE REVIEW

File: config/lang-scheme.yml [hunk 1]
```diff
@@ -1,93 +1,35 @@
...
-            (newline)
-            (print-lines (rest lines)))
-          (values)))
-
-      (let* ([lines (read-lines)]
-             [processed (map process lines)])
-        (print-lines processed))
-    stdin:  "1\n2\n3\n"
-    stdout: "\\A2\n4\n6\n\\z"
+      (begin (display "hi from Scheme") (newline))
```
Comment (simonista) [inline config/lang-scheme.yml]: it would be nice to keep this simple test the same as all the other languages

File: config/lang-scheme.yml [hunk 2]
```diff
@@ -1,93 +1,35 @@
...
 filename: source.scm
 docker_image: straitjacket-scheme
 
 file_extensions:
-  - rkt
   - scm
-  - scheme
 
 tests:
-  template:
```
Comment (simonista) [inline config/lang-scheme.yml]: we still need a template

File: config/lang-racket.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+name: scheme
```
Comment (javawizard) [inline config/lang-racket.yml]: Womp

File: config/lang-scheme.yml (hunk 1 above)
Comment (dented42) [inline config/lang-scheme.yml]: "hi from guile"? Very well, just a sec.

File: config/lang-scheme.yml (hunk 2 above)
Comment (dented42) [inline config/lang-scheme.yml]: I moved the old racket template to guile, removed the square brackets, and included the module that is supposed to have `read-line`.

Also made the racket template much shorter.

File: config/lang-racket.yml (hunk 3 above)
Comment (dented42) [inline config/lang-racket.yml]: Let us never speak of this again.

File: config/lang-scheme.yml [hunk 4]
```diff
@@ -43,51 +41,25 @@ tests:
...
-            (newline)
-            (print-lines (rest lines)))
-          (values)))
-
-      (let* ([lines (read-lines)]
-             [processed (map process lines)])
-        (print-lines processed))
-    stdin:  "1\n2\n3\n"
-    stdout: "\\A2\n4\n6\n\\z"
+      (begin (display "hi from guile") (newline))
```
Comment (simonista) [inline config/lang-scheme.yml]: any reason not to keep this as the "multiple all inputs by 2" test?

Comment (dented42) [inline config/lang-scheme.yml]: I wanted to make as few changes as possible. The multiply-by-2 test was originally for scheme, and the hi-from-guile test was originally for guile. Since scheme was renamed racket and and guile renamed scheme, git makes the mistake of thinking that the-from-guile test replaced the multiply-by-2 test; git, rather unfortunately, does not keep track of renames and therefore doesn't realise that this is a different file with the same name.

That being said, if changing it to the multiply-by-2 test is desirable, I can't think of a good reason not to. :)

Comment (simonista) [inline config/lang-scheme.yml]: Oh, I didn't notice that the previous guile file was also lacking.  This makes sense.

File: config/lang-scheme.yml [hunk 5]
```diff
@@ -1,22 +1,20 @@
...
-  - rkt
   - scm
-  - scheme
 
 tests:
   template:
     source: |
-      #lang racket
-
+      (use-modules (ice-9 rdelim))
```
Comment (codekitchen) [inline config/lang-scheme.yml]: this template fails for me, does it need to be adapted now that this file is for guile/scheme rather than racket?

```
    Backtrace:
    In ice-9/boot-9.scm:
     157: 13 [catch #t #<catch-closure 1956660> ...]
    In unknown file:
       ?: 12 [apply-smob/1 #<catch-closure 1956660>]
    In ice-9/boot-9.scm:
      63: 11 [call-with-prompt prompt0 ...]
    In ice-9/eval.scm:
     432: 10 [eval # #]
    In ice-9/boot-9.scm:
    2401: 9 [save-module-excursion #<procedure 1973a00 at ice-9/boot-9.scm:4045:3 ()>]
    4050: 8 [#<procedure 1973a00 at ice-9/boot-9.scm:4045:3 ()>]
    1724: 7 [%start-stack load-stack ...]
    1729: 6 [#<procedure 198bfc0 ()>]
    In unknown file:
       ?: 5 [primitive-load "/src/source.scm"]
    In ice-9/eval.scm:
     432: 4 [eval # #]
     387: 3 [eval # #]
     386: 2 [eval #<memoized (first <0>)> (("1" "2" "3") . #<directory # 194ac60>)]
     393: 1 [eval #<memoized first> (("1" "2" "3") . #<directory (guile-user) 194ac60>)]
    In unknown file:
       ?: 0 [memoize-variable-access! #<memoized first> #<directory # 194ac60>]

    ERROR: In procedure memoize-variable-access!:
    ERROR: Unbound variable: first
```

Comment (dented42) [inline config/lang-scheme.yml]: Yes, kind of. The code itself works in standard R5RS, which guile implements. But the `read-line` function is not a part of R5RS. It's commonly provided with most scheme implementations, and I Googled how to get it in guile, and I guess I made a mistake in the line that's supposed to load it into the current namespace.

TL;DR I done goofed at requiring/importing `read-line`.

Comment (dented42) [inline config/lang-scheme.yml]: Oh whoops. Whoever wrote this used the rackety `first` and `rest` functions instead of the traditional lispy `car` and `cdr`. A bit odd because you'd think they'd have also used `empty?` instead of `null?`. Oh well, hang on a sec.

**Comment by codekitchen** [comment]
Cool thanks, I squashed this down to one commit and merged it in.
